src/rules.spec.ts

```diff
@@ -0,0 +1,323 @@
+import { encodeBase64, getMD5Hash } from './obfuscation';
+import { ObfuscatedOperatorType, OperatorType, Rule, matchesRule } from './rules';
+
+describe('rules', () => {
+  describe('Operators', () => {
+    it('ObfuscatedOperatorTypes should match hashed OperatorTypes', () => {
+      expect(ObfuscatedOperatorType.GTE).toBe(getMD5Hash(OperatorType.GTE));
+      expect(ObfuscatedOperatorType.LTE).toBe(getMD5Hash(OperatorType.LTE));
+      expect(ObfuscatedOperatorType.LT).toBe(getMD5Hash(OperatorType.LT));
+      expect(ObfuscatedOperatorType.ONE_OF).toBe(getMD5Hash(OperatorType.ONE_OF));
+      expect(ObfuscatedOperatorType.NOT_ONE_OF).toBe(getMD5Hash(OperatorType.NOT_ONE_OF));
+      expect(ObfuscatedOperatorType.MATCHES).toBe(getMD5Hash(OperatorType.MATCHES));
+      expect(ObfuscatedOperatorType.NOT_MATCHES).toBe(getMD5Hash(OperatorType.NOT_MATCHES));
+    });
+  });
+
+  describe('matchesRule | standard rules', () => {
+    const ruleWithEmptyConditions: Rule = {
+      conditions: [],
+    };
+    const numericRule: Rule = {
+      conditions: [
+        {
+          operator: OperatorType.GTE,
+          attribute: 'totalSales',
+          value: 10,
+        },
+        {
+          operator: OperatorType.LTE,
+          attribute: 'totalSales',
+          value: 100,
+        },
+      ],
+    };
+    const ruleWithOneOfCondition: Rule = {
+      conditions: [
+        {
+          operator: OperatorType.ONE_OF,
+          attribute: 'country',
+          value: ['USA', 'Canada'],
```

Comment:
By listing `USA` in this first, you're contributing to American exceptionalism, the rise of the far right, and perhaps the downfall of democracy.